R/pmap.R

```diff
@@ -101,11 +101,14 @@ pmap_chr <- function(.l, .f, ..., .progress = FALSE) {
 
 pmap_ <- function(.type, .l, .f, ..., .progress = FALSE, ..error_call = caller_env()) {
   .f <- as_mapper(.f, ...)
-  if (is.data.frame(.l)) {
-    .l <- as.list(.l)
-  }
-  i <- 0L
 
+  .l <- vctrs_list_compat(.l)
+  vec_check_list(.l, call = ..error_call)
+
+  .l <- map(.l, vctrs_list_compat)
+  .l <- vec_recycle_common(!!!.l, .arg = ".l", .call = ..error_call)
```

Comment:
I'm going to make the assumption that recycling in the map2/pmap case really isn't very expensive, because most people won't be doing `map2(1:1000, 1L)` - i.e. where one of the inputs actually needs recycling.

If they are doing that, they probably should have just supplied `.y` as a fixed argument

(This is compared to what we currently do, which is _don't_ recycle but instead use `x[[1L]]` repeatedly)

Reply:
Yeah, I'm also assuming vctrs will use ALTREP for recycling one day, making the cost even lower.